### PR TITLE
job id in status

### PR DESCRIPTION
**Pull request by galak**
Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

File: scripts/check_compliance.py [hunk 1]
```diff
@@ -947,7 +947,7 @@ def set_commit_status(status, msg):
                              "pylint"}:
                 msg = "```\n" + msg + "\n```"
             fail_msg += msg + "\n\n"
-            set_commit_status('failure', 'Checks failed')
+            set_commit_status('failure', 'Checks failed on run ' + build_number)
```
Comment (ulfalizer) [inline scripts/check_compliance.py]: maybe `set_commit_status()` could be updated instead:

```python
    def set_commit_status(status, msg):
        # 'case' gets set in the loop.
        # pylint: disable=undefined-loop-variable
        github_commit.create_status(
            status, name2doc[case.name],
            "{} ({})".format(msg, build_number), case.name)
```

Then the build number should get included for all types of statuses.

File: scripts/check_compliance.py [hunk 2]
```diff
@@ -56,7 +56,6 @@ def git(*args):
 # The absolute path of the top-level git directory
 GIT_TOP = git("rev-parse", "--show-toplevel").strip()
 
-
```
[the remark this answers is not in the document]
Comment (galak) [reply scripts/check_compliance.py]: fixed

File: scripts/check_compliance.py (hunk 1 above)
Comment (galak) [reply scripts/check_compliance.py]: DOne.

File: scripts/check_compliance.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ulfalizer) [reply scripts/check_compliance.py]: Still seems to be removed.

File: scripts/check_compliance.py [hunk 3]
```diff
@@ -947,7 +948,7 @@ def set_commit_status(status, msg):
                              "pylint"}:
                 msg = "```\n" + msg + "\n```"
             fail_msg += msg + "\n\n"
-            set_commit_status('failure', 'Checks failed')
+            set_commit_status('failure', 'Checks failed on run')
```
Comment (ulfalizer) [inline scripts/check_compliance.py]: Accidental leftover "on run".

File: scripts/check_compliance.py [hunk 4]
```diff
@@ -1085,6 +1087,12 @@ def init_github(args):
     else:
         github_pr = None
 
+    # Get the shippable build number, useful to find logs
+    build_number = os.environ.get('BUILD_NUMBER')
```
Comment (ulfalizer) [inline scripts/check_compliance.py]: Maybe double quotes just to be consistent with next lines (though this script is a bit inconsistent in general).

File: scripts/check_compliance.py (hunk 3 above)
Comment (galak) [reply scripts/check_compliance.py]: fixed

File: scripts/check_compliance.py (hunk 4 above)
Comment (galak) [reply scripts/check_compliance.py]: done

File: scripts/check_compliance.py [hunk 5]
```diff
@@ -928,8 +929,9 @@ def report_test_results_to_github(suite):
     def set_commit_status(status, msg):
         # 'case' gets set in the loop.
         # pylint: disable=undefined-loop-variable
-        github_commit.create_status(status, name2doc[case.name], msg,
-                                    case.name)
+        github_commit.create_status(status, name2doc[case.name],
+                "{} (build no. {})".format(msg, build_number),
+                case.name)
```
Comment (ulfalizer) [inline scripts/check_compliance.py]: Could reformat this as

```python
        github_commit.create_status(
                status, name2doc[case.name],
                "{} (build no. {})".format(msg, build_number),
                case.name)
```

If there are arguments directly after `(`, then the rest of the arguments aren't usually less indented.

Comment (galak) [reply scripts/check_compliance.py]: Done.

**Pull request merged**